File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Beginners/Pages/BingMainPage/BingMainPage.cs
using OpenQA.Selenium;

namespace PatternsInAutomatedTests.Beginners.Pages.BingMainPage
{
    public class BingMainPage
    {
        private readonly IWebDriver browser;
        private readonly string url = @"http://www.bing.com/";

        public BingMainPage(IWebDriver browser)
        {
            this.browser = browser;
        }

        protected BingMainPageElementMap Map
        {
            get
            {
                return new BingMainPageElementMap(this.browser);
            }
        }

        public BingMainPageValidator Validate()
        {
            return new BingMainPageValidator(this.browser);
        }

        public void Navigate()
        {
            this.browser.Navigate().GoToUrl(this.url);
        }

        public void Search(string textToType)
        {
            this.Map.SearchBox.Clear();
            this.Map.SearchBox.SendKeys(textToType);
            this.Map.GoButton.Click();
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/pageobject/Pages/BingMainPage/BingMainPage.java
package pageobject.pages.bingmainpage;

import org.openqa.selenium.WebDriver;

public class BingMainPage {
    private final WebDriver browser;
    private final String url = "http://www.bing.com/";

    public BingMainPage(WebDriver browser) {
        this.browser = browser;
    }
    protected BingMainPageElements elements() {
        return new BingMainPageElements(browser);
    }

    public BingMainPageAssertions assertions() {
        return new BingMainPageAssertions(browser);
    }

    public void navigate() {
        browser.navigate().to(url);
    }

    public void search(String textToType) {
        elements().searchBox().clear();
        elements().searchBox().sendKeys(textToType);
        elements().goButton().click();
    }
}
